File: src/Web/WebUI/src/app/core/models/maintainer/submenu.model.ts
export interface SubMenuDto{
    subMenuId: string;
    subMenuName: string;
    menuName: string;
    isActive: boolean;
}

export interface SubMenuVM{
    subMenuId: string;
    subMenuName: string;
    menuId: string;
}

export interface CreateSubMenu{
    subMenuName: string;
    menuId: string;
}

export interface UpdateSubMenu{
    subMenuId: string;
    subMenuName: string;
    menuId: string;
}

File: src/Application/Common/Models/SubMenuModel.cs
namespace Application.Common.Models;

public record CreateSubMenu
{
    public string SubMenuName { get; init; } = default!;
    public int SubMenuOrder { get; init; }
}

public record UpdateSubMenu
{
    public Guid SubMenuId { get; set; }
    public string SubMenuName { get; set; } = default!;
    public int SubMenuOrder { get; set; }
}